### PR TITLE
fix(neo4j backend): Query synatx Error in Neo4jGraphService.findRelatedUrns 

### PR DESCRIPTION
**Pull request by amorskoy**
Please see linked issue for detailed description 
https://github.com/linkedin/datahub/issues/2678

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/graph/Neo4jGraphService.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public void addEdge(@Nonnull final Edge edge) {
 
     String statementString =
         String.format(matchTemplate, sourceType, srcCriteria, relationshipTypeFilter, edgeCriteria,
-            destinationType, destCriteria);
+            destinationType, destCriteria, sourceType);
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/graph/Neo4jGraphService.java:102]: Should we be returning destinationType?

Comment (amorskoy) [reply metadata-io/src/main/java/com/linkedin/metadata/graph/Neo4jGraphService.java:102]: @jjoyce0510 I believe you are right. At this point I am unable to recall, what was the reason for `sourceType` here versus `destinationType`. The `destinationType` looks expected. Let me change it

Comment (amorskoy) [reply metadata-io/src/main/java/com/linkedin/metadata/graph/Neo4jGraphService.java]: Done

**Comment by shirshanka** [comment]
@EnricoMi : do your latest PR-s remove the need for this PR?

**Comment by EnricoMi** [comment]
@shirshanka I am using the following patch in #3011: https://github.com/linkedin/datahub/issues/2678#issuecomment-901117062

To be honest, this issue should be fixed separate from #3011 though.

**Comment by maggiehays** [comment]
Hey @dexter-mh-lee! This one has been sitting for a good while - please take a look to see if this is still an issue - let's address it if so!

**Comment by EnricoMi** [comment]
I'd say that has been resolved by merging #3011.

**Comment by jjoyce0510** [comment]
Resolved via https://github.com/datahub-project/datahub/pull/3011.

Closing this PR.

Thank you for the inspiration @amorskoy !

**Pull request closed without merging**